### PR TITLE
Add a doesNotStartWith option to the generate function. 

### PR DESCRIPTION
**Pull request by natashad**
We want to be able to generate a SINs that don't start with specific numbers for testing reasons.

### CODE REVIEW

**Review by adyblenko — APPROVED** [review]
LGTM just a small suggestion

File: social-insurance-number.js [hunk 1]
```diff
@@ -25,11 +25,7 @@
 
   SocialInsuranceNumber.generate = function(options) {
     options = options || {};
-    var startsWith = options.startsWith;
-    if (!startsWith) {
-      var province = options.province || randomChoice(Object.keys(PROVINCES));
-      startsWith = randomChoice(PROVINCES[province]);
-    }
+    var startsWith = options.startsWith || getStartsWithIfNotSet(options.doesNotStartWith, options.province);
```
[the remark this answers is not in the document]
Comment (natashad) [reply social-insurance-number.js:28]: 👍  That makes sense, I'll update

[the remark this answers is not in the document]
Comment (natashad) [reply social-insurance-number.js]: @adyblenko -- Updated, I've changed the logic to try and respect province if possible and error out if there are no valid choices. Not sure if my error message is the best, happy to hear alternative options.

**Review by adyblenko — APPROVED** [review]
very nice and clean - cheers 🎉

**Pull request merged**